src/CircuitOptimizer.cpp

```diff
@@ -527,4 +527,107 @@ namespace qc {
 		// remove resulting identities from circuit
 		removeIdentities(qc);
 	}
+	void CircuitOptimizer::removeFinalMeasurements(QuantumComputation &qc)
+	{
+		//remove final measurements in the circuit (for use in mapping tool)
+		auto it = qc.ops.rbegin();
+		while (it != qc.ops.rend())
+		{
+			if ((*it)->isNonUnitaryOperation())
+			{
+				if ((*it)->getType() == Measure)
+				{
+					qc.ops.erase((++it).base());
+				}
+				else
+				{
+					//First non measurement
+					break;
+				}
+			}
+			else
+			{ //not a measurement
+				break;
+			}
+		}
```

Comment:
This does not consider compound operations (which might contain measurements at the end).
Additionally it does not work in case a measurement on one qubit is followed by an operation on another qubit, e.g.
```qasm
OPENQASM 2.0;
include "qelib1.inc";
qreg q[2];
creg c[2];
measure q[0]->c[0];
x q[1];
```
An approach similar to the `removeDiagonalGatesBeforeMeasure` pass should be adopted here. It builds a DAG representation of the circuit which should then allow to just erase the "endpoints" if they are measurements.